derp/derp_server.go

```diff
@@ -1338,6 +1466,17 @@ type sclient struct {
 	// client that it's trying to establish a direct connection
 	// through us with a peer we have no record of.
 	peerGoneLim *rate.Limiter
+
+	// flows is a cache of the subset of flows statistics for
+	// which this client is either the src or the mesh peer
+	// forwarding for the src. The flowStats fields are changed by
+	// only the sclient read loop, but the flowStats must be
+	// allocated by the server.
+	flows map[key.NodePublic]map[key.NodePublic]*flowStats
+
+	// flowsCleanNow is set by the server to indicate it is time
+	// for this sclient to look for and free inactive flows.
+	flowsCleanNow bool
```

Comment:
this is guarded by what? s.mu also? It doesn't look consistently guarded elsewhere. flowsTimerFunc holds s.mu but cleanFlows reads it without any locking.